Aptoide/aptoide-client-v8
# [ASV-1663] Download state after 100%

**What does this PR do?**

   Initially this PR aimed at fixing the download staying at 100% problem, but since we could not reproduce that issue with a specific behavior, we just added logs so that the next time it happens we can correctly trace the issue.
In this PR we also took the chance to look into other problems related with the downloads and tried to optimize some flows. The other fixes were:
 - Checking if file is already in the final path before trying to move it.
 - Optimized install chain to remove unnecessary database access and verifications.
 - Fixed download state parsing (initial state was not being correctly parsed).
 - Attempted at fixing a problem where we wouldn't be able to resume or cancel a download because the app downloader object was null (added logs since I couldn't reproduce it no more. I think the fix of the previous line could have been the solution)
 - Fixed the state after the installing state ( download app, at the end it will go through the installing state and pop the install dialog, cancel. When you come back press the install button again and there, the pause button would disappear).

As a result of this PR we also created the following ticket [ASV-1704](https://aptoide.atlassian.net/browse/ASV-1704)

**Database changed?**

   No

**Where should the reviewer start?**

- [ ] AptoideDownloadManager.java


**How should this be manually tested?**

  Since there are a lot of fixes on the downloads I would recommend the following:
 - Perform a download, pause and resume (to try to force the appdownloader problem).
 - Check if the installing status is being correctly handled in both APPC apps and non appc apps (also check the state after the install)
 - See if there are no problems popping the install dialog, besides the problem that we could not reproduce in a consistent way, where the download would be stuck at 100% (if you get this, please call me to check the logs).
 

**What are the relevant tickets?**

  Tickets related to this pull-request: [ASV-1663](https://aptoide.atlassian.net/browse/ASV-1663)

**Questions:**

   Does this add new dependencies which need to be added? (Eg. new keys, etc.) 



**Code Review Checklist**
